File: Backend/src/BaboCare.Application/Dtos/Activities/ActivityDtos.cs
using BaboCare.Domain.Entities.Activities;

namespace BaboCare.Application.Dtos.Activities;

/// <summary>
/// 建立寶寶活動的請求 DTO
/// </summary>
public record CreateActivityRequestDto(
    string BabyId,
    ActivityType Type,
    string PhotoUrl,
    string? Notes,
    int? TypeOption
);

/// <summary>
/// 活動的回應 DTO（用於動態牆與轉盤）
/// </summary>
public record ActivityResponseDto(
    string Id,
    string BabyId,
    string BabyName,
    string? BabyAvatarUrl,
    ActivityType Type,
    string PhotoUrl,
    string? Notes,
    int? TypeOption,
    DateTime CreatedAt,
    string CreatedByUserId
);

/// <summary>
/// 動態牆頭像摘要（用於頂部頭像列）- 依最新活動排序
/// </summary>
public record BabyActivitySummaryDto(
    string BabyId,
    string BabyName,
    string? BabyAvatarUrl,
    DateTime LatestActivityAt,
    int ActivityCount
);

File: Frontend/src/api/dtos/Activities/activityDtos.ts
// DTO Types for Activity management

export const ActivityType = {
  Feeding: 1,
  Eating: 2,
  Diaper: 3,
  Sleep: 4,
  Mood: 5,
} as const;

export type ActivityTypeValue = typeof ActivityType[keyof typeof ActivityType];

export const EatingOption = {
  Supplementary: 1,
  Lunch: 2,
} as const;

export type EatingOptionValue = typeof EatingOption[keyof typeof EatingOption];

export const MoodOption = {
  Outing: 1,
  Playing: 2,
  Unhappy: 3,
} as const;

export type MoodOptionValue = typeof MoodOption[keyof typeof MoodOption];

export interface IActivity {
  id: string;
  babyId: string;
  babyName: string;
  babyAvatarUrl?: string;
  type: ActivityTypeValue;
  photoUrl: string;
  notes?: string;
  typeOption?: number;
  createdAt: string;
  createdByUserId: string;
}

export interface ICreateActivityRequest {
  babyId: string;
  type: ActivityTypeValue;
  photoUrl: string;
  notes?: string;
  typeOption?: number;
}

export interface IBabyActivitySummary {
  babyId: string;
  babyName: string;
  babyAvatarUrl?: string;
  latestActivityAt: string;
  activityCount: number;
}
